addon/services/ember-elsewhere.js

```diff
@@ -16,24 +17,31 @@ export default Service.extend({
     };
     this._schedule();
   },
+
   clear(sourceId) {
     delete this._alive[sourceId];
     this._schedule();
   },
+
   _schedule() {
     run.scheduleOnce('afterRender', this, this._process);
   },
+
   _process() {
     let newActives = {};
     let alive = this._alive;
 
-    Object.keys(alive).forEach(sourceId => {
+    Object.keys(alive).forEach((sourceId) => {
       let { target, component, order } = alive[sourceId];
-      let existing = newActives[target];
-      if (!existing || existing.order < order) {
-        newActives[target] = component ? { component, order } : null;
-      }
+      newActives[target] = newActives[target] || emArray();
+      let newActive = component ? { component, order } : null;
+
+      newActives[target].addObject(newActive);
```

Comment:
Plain Array `push` should be sufficient here, instead of `addObject`.

We are deliberately not attempting to mutate existing lists in a way that is observable from the outside -- we're computing brand new lists each time `_process` runs. So observability is not a concern.

And the de-duplication that `addObject` does will never find any matches, since we are always creating a new `newActive` object in each iteration of the loop.